File: JANUARY/24-01-2026/1877.c
int cmp(const void* a, const void* b){
    return (*(int*)a - *(int*)b);
}

int minPairSum(int* nums, int numsSize){
    qsort(nums, numsSize, sizeof(int), cmp);
    int i = 0, j = numsSize - 1, ans = 0;
    while (i < j) {
        int s = nums[i] + nums[j];
        if (s > ans) ans = s;
        i++;
        j--;
    }
    return ans;
}

File: JANUARY/24-01-2026/1877.swift
class Solution {
    func minPairSum(_ nums: [Int]) -> Int {
        let a = nums.sorted()
        var i = 0, j = a.count - 1, ans = 0
        while i < j {
            ans = max(ans, a[i] + a[j])
            i += 1
            j -= 1
        }
        return ans
    }
}